File: 10/main.cpp
#include <iostream>
#include <cstdint>
using namespace std;

int main() {
	const int NUM_PRIMES = 1000;
	int primes[NUM_PRIMES];
	primes[0] = 2;
	int count = 1;
	int i = 3;
	while (count < NUM_PRIMES) {
		for (int j = 2; j < i; j++) {
			if (j >= 2000000) {
				uint64_t sum = 0;
				for (int i = 0; i < count; i++)
				{
					sum += primes[count];
				}
				cout << sum << endl;
				// Done, found desired sum
				exit(0);
			}
			if (i % j == 0) {
				break;
			}
			if (j == i-1) {
				primes[count] = j+1;
				count++;
			}
		}
		i++;
	}
	cout << primes[count] << endl;
}

File: 10/main.go
package main

import (
	"fmt"
	"math/big"
	"os"
)

func main() {
	const NUM_PRIMES int = 1000001
	primes := make([]int, NUM_PRIMES);
	primes[0] = 2;
	i := 3
	for count := 1; count < NUM_PRIMES; {
		for j := 2; j < i; j++ {
			// Want sum of all primes < 2,000,000
			if j >= 2000000 {
				sum := big.NewInt(0);
				for prime := range primes {
					sum.Add(sum, big.NewInt(int64(prime)));
				}
				fmt.Println(sum);
				os.Exit(0);
			}
			if i % j == 0 {
				break;
			}
			if j == i-1 {
				primes[count] = j+1;
				count++;
			}
		}
		i++;
	}
	fmt.Println(primes);
}
